src/Agent.cs

```diff
@@ -62,7 +62,8 @@ public Agent(String apiKey)
       _collector = new Collector(apiKey);
 
       var validationOptions = RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.ExplicitCapture;
-      _validateMetric = new Regex(@"^([\d\w\-_]+\.)*[\d\w\-_]+$", validationOptions);
+      _validateMetric = new Regex(@"^([a-zA-Z0-9\-_]+\.)*[a-zA-Z0-9\-_]+$", validationOptions);
```

Comment:
Just for funsies, you should add some tests that have `_` and `-` in the metric name, and also some numbers.